ros2_batch_job/__main__.py

```diff
@@ -505,10 +505,20 @@ def run(args, build_function, blacklisted_package_names=None):
 
     # Now inside of the workspace...
     with change_directory(args.workspace):
+        def need_package_from_pipy(pkg_name):
+            try:
+                importlib.import_module(pkg_name)
+            except ModuleNotFoundError:
+                return True
+
+            return False
+
         print('# BEGIN SUBSECTION: install Python packages')
-        # Update setuptools
-        job.run(['"%s"' % job.python, '-m', 'pip', 'install', '-U', 'pip', 'setuptools'],
-                shell=True)
+
+        # Update setuptools if it is not already available.
+        if need_package_from_pipy('setuptools'):
+            job.run(['"%s"' % job.python, '-m', 'pip', 'install', '-U', 'pip', 'setuptools'],
+                    shell=True)
```

Comment:
I think that this is actually dead code without MacOS and we should consider removing it entirely.

* Ubuntu is installing setuptools from apt during image build.
* RHEL in installing setuptools from dnf during image build.
* Windows is installing setuptools from pip (via chef) during image build.

So only macOS lacks any kind of per-job container (aside from a virtualenv) where setuptools would need to be updated each build in order to stay current.

Reply:
It's totally fine by me to remove it.

Reply:
Done in 82c018c